### ISSUE TITLE
[java] AvoidArrayLoops false negative with do-while loops

### ISSUE THREAD

**Issue by Zustin** [pmd/pmd#4091]
**Affects PMD Version:6.47.0** 

**Rule:AvoidArrayLoops**

Please provide the rule name and a link to the rule documentation:
<https://pmd.github.io/latest/pmd_rules_java_performance.html#avoidarrayloops>

**Description:**
Hi, I found an inconsistent bug about the rule `AvoidArrayLoops`, please read the code example below. PMD reports a warning in line 10, but no warning in line 6. However, these two statements are equivalent. Hence, I think this is an inconsistent outcome. The root cause is the xpath pattern in [Link](https://pmd.github.io/latest/pmd_rules_java_performance.html#avoidarrayloops) doesn't consider `DoStatement`. Thanks.

**Code Sample demonstrating the issue:**
```java
public void foo() {
    int[] a = new int[10];
    int[] b = new int[10];
    final int c = 6;
    do {
        b[i] = a[i+6];  // no warning
        i++;
    } while(i < 6);
    while(i < 6) {
        b[i] = a[i + 6];  // report a warning
        i++;
    }
}
```

**Running PMD through:** *[Maven]*

**Comment by techrajdeep** [comment]
I am looking for an issue for my first contribution . Can you help me here. Can I take on this .

**Comment by adangel** [comment]
> I am looking for an issue for my first contribution . Can you help me here. Can I take on this .

@techrajdeep: This issue seems already be taken on by @yasarshaikh

**Comment by mkorzeniewski** [comment]
Is this issue still unavailable?

**Comment by jsotuyod** [comment]
@mkorzeniewski we appreciate your intention to help, but yes, we have a PR from another user en route to be merged.

If you want to contribute however, there are a few other easy issues laying around [here](https://github.com/pmd/pmd/labels/good%20first%20issue)
